### ISSUE TITLE
"Get DNS name servers from OS configuration" fails with UDP-only router

### ISSUE THREAD

**Issue by lusitania** [lieser/dkim_verifier#42]
Hi

This is similar to "Get DNS name servers form OS configuration" #24 but for Windows.

If I remove the default "DNS name servers" (which by the way is a redundant way of saying it) messages don't get verified at all. The error message I then get is "Error connecting to DNS server". Same is true if I set the fields value to *127.0.0.1*. I therefore assume that the DNS address provided through the OS isn't queried/used at all.

Since I'm often roaming I can't know the DNS for all networks I am connected to beforehand. This issue is a blocker for me.

Best

Edit: I just set the field to my current local DNS forwarder and I still get the error (but not with Google's DNS). I suppose that there is more to this. Is there a log I can look up and send you?

**Comment by lieser** [comment]
To see thats going on enable debugging under options > advanced.

The option `Get DNS name servers form OS configuration` does not mean that the DNS resolver provided by the OS is used. It just means that the DNS resolver shipped with the add-on is first trying to use the DNS servers that are configured to be used in the OS.

Because you say it's still not working then you set the DNS server manually the problem is probably the connecting to the DNS server, not getting which to use form the OS configuration.

A possible problems could be that the DNS server you are trying to use only supports UDP, not TCP (more information about it at [https://github.com/lieser/dkim_verifier/wiki/DNS#javascript-dns-library](https://github.com/lieser/dkim_verifier/wiki/DNS#javascript-dns-library)).

Note that as a workaround you could try to use the [libunbound resolver](https://github.com/lieser/dkim_verifier/wiki/DNS#libunbound)

**Comment by lusitania** [comment]
Okay, so I've set the field value to the DNS name my router is connected and it works. So my router runs a UDP only DNS proxy which otherwise would be perfectly fine. At home this is no issue because I was able to look up the value and it probably will not change. While roaming, though, I don't think this will work.

DNSSEC/TLSA Validator is not compatible with the current Thunderbird batch (31.4.0). It refuses to install. So this option is out, too.

(Found the debug log, thanks)
